libs/ui/elements/src/lib/record-api-form/record-api-form.component.html

```diff
@@ -37,15 +37,37 @@
           </div>
         </div>
       </div>
-      <div class="flex flex-col gap-3">
-        <p class="text-sm" translate>record.metadata.api.form.offset</p>
-        <gn-ui-text-input
-          class="w-20"
-          [value]="offset$ | async"
-          (valueChange)="setOffset($event)"
-          hint=""
-        >
-        </gn-ui-text-input>
+      <div class="flex flex-col gap-3 relative">
+        <p class="text-sm" [class.text-gray-600]="!supportOffset" translate>
+          record.metadata.api.form.offset
+        </p>
+        <div class="flex items-center">
+          <gn-ui-text-input
+            class="w-20"
+            [value]="offset$ | async"
+            [extraClass]="
+              supportOffset ? '' : 'text-gray-600 bg-gray-200 opacity-70'
+            "
+            (valueChange)="supportOffset ? setOffset($event) : null"
+            hint=""
+          >
+          </gn-ui-text-input>
+          <div
+            *ngIf="!supportOffset"
+            class="flex items-center gap-2 text-red-500 z-10 ml-3 cursor-pointer"
+          >
+            <span
+              class="material-symbols-outlined"
+              matTooltip="Not supported on this service"
+            >
+              warning
+            </span>
+          </div>
+        </div>
+        <div
+          *ngIf="!supportOffset"
+          class="absolute inset-0 bg-transparent cursor-not-allowed"
+        ></div>
```

Comment:
I'm not sure this is really necessary; if the text input is disabled and the warning icon shows up with a mat-tooltip that should be enough in terms of feedback

Reply:
The text input can't be directly disabled, but this provides an alternative to prevent user interaction.